src/pip/_internal/index.py

```diff
@@ -216,6 +216,11 @@ def sort_path(path):
                             sort_path(os.path.join(path, item))
                     elif is_file_url:
                         urls.append(url)
+                    else:
+                        logger.warning(
+                            "Path '{0}' is ignored: "
+                            "it is a directory.".format(url),
```

Comment:
Nit: this should be `path` (though they should be equal), right?

Reply:
Gosh, uh, I… guess? Since the outer condition specifies `is_local_path or is_file_url`, but the one above takes `is_file_url`, so this leaves only `is_local_path and not is_file_url`, which would have triggered `path = url`. It’s so difficult to keep this straight.